rpc/amqp-rpc.go

```diff
@@ -95,6 +97,16 @@ func AMQPDeclareExchange(conn *amqp.Connection) error {
 	return err
 }
 
+// Create a quick consumer ID we can recreate later if we need to cancel
+// a consumer.
+func getConsumerName(serverQueue string) (string, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s.%s", serverQueue, hostname), nil
```

Comment:
from the AMQP [docs](https://godoc.org/github.com/streadway/amqp#Channel.Consume):

```
The consumer is identified by a string that is unique and scoped for all consumers on this channel.
If you wish to eventually cancel the consumer, use the same non-empty idenfitier in Channel.Cancel.
```

I'm currently completely redoing parts of this, this being one. I think the best thing here to just generate some kind of unique-ish identifier to append to the server queue name in order to reduce conflicts.